labelbox/schema/dataset.py

```diff
@@ -425,3 +427,28 @@ def export_data_rows(self, timeout_seconds=120) -> Generator:
             logger.debug("Dataset '%s' data row export, waiting for server...",
                          self.uid)
             time.sleep(sleep_time)
+
+    def _convert_metadata_field(self, metadata_field):
+        if isinstance(metadata_field, DataRowMetadataField):
+            return metadata_field
+        elif isinstance(metadata_field, dict):
+            if not all(key in metadata_field for key in ("schema_id", "value")):
+                raise ValueError(
+                    f"Custom metadata field '{metadata_field}' must have 'schema_id' and 'value' keys"
+                )
+            return DataRowMetadataField(schema_id=metadata_field["schema_id"],
+                                        value=metadata_field["value"])
+        else:
+            raise ValueError(
+                f"Metadata field '{metadata_field}' is neither 'DataRowMetadataField' type or a dictionary!"
+            )
+
+    def _parse_metadata(self, metadata_fields):
+        # Convert all metadata fields to DataRowMetadataField type
+        metadata_fields = [
+            self._convert_metadata_field(m) for m in metadata_fields
+        ]
+        mdo = self.client.get_data_row_metadata_ontology()
+        parsed_metadata = list(
+            chain.from_iterable(mdo.parse_upsert(m) for m in metadata_fields))
```

Comment:
I feel like a function for parsing multiple metadata items belongs on the metadata ontology object. E.g. if you wanted to use this function on the data row class it would have to be duplicated.